File: signature/backbonejs.go
package signature

var BackboneJsSignature = Signature{
	Name:        "Backbone.js",
	Description: "Backbone.js is a JavaScript rich-client web app framework based on the model–view–controller design paradigm, intended to connect to an API over a RESTful JSON interface.",
	Cpe:         "cpe:/a:backbone_project:backbone",

	DetectPattern: DetectPattern{
		Bodies: []string{
			`Backbone.history has already been started`,
			`t.VERSION="(\d+\.\d+\.\d+)"`,
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`t.VERSION="(\d+\.\d+\.\d+)"`,
		},
	},
}

File: src/signatures/technologies/backbone_js.ts
import type { Signature } from "../_types.js";

export const backboneJsSignature: Signature = {
  name: "Backbone.js",
  description:
    "BackboneJS is a JavaScript library that allows to develop and structure the client side applications that run in a web browser.",
  cpe: "cpe:/a:backbone_project:backbone",
  rule: {
    confidence: "high",
    urls: ["backbone.*\\.js"],
    javascriptVariables: {
      Backbone: "",
      "Backbone.VERSION": "^(.+)$",
    },
  },
};
